api/src/main/java/me/lokka30/treasury/api/economy/currency/CurrencyManager.java

```diff
@@ -0,0 +1,248 @@
+     *
+     * @param currencyCharacter currency character to check if busy
+     * @return boolean value
+     * @author MrIvanPlays
+     * @since v1.0.0
+     */
+    public boolean hasCurrency(@NotNull Character currencyCharacter) {
+        Objects.requireNonNull(currencyCharacter, "currencyCharacter");
+        return BY_CHAR.containsKey(currencyCharacter);
+    }
+
+    /**
+     * Returns the {@link Currency} bound to the specified {@link UUID} {@code currencyId}, if present.
+     *
+     * @param currencyId currency id you want the currency of
+     * @return currency if present, empty optional otherwise
+     * @author MrIvanPlays
+     * @since v1.0.0
+     */
+    public Optional<Currency> getCurrency(@NotNull UUID currencyId) {
+        Objects.requireNonNull(currencyId, "currencyId");
+        return Optional.ofNullable(BY_ID.get(currencyId));
+    }
+
+    /**
+     * Returns the {@link Currency} bound to the specified {@link String} {@code currencyName}, if present.
+     *
+     * @param currencyName the name of the currency needed
+     * @return currency if present, empty optional otherwise
+     * @author MrIvanPlays
+     * @since v1.0.0
+     */
+    public Optional<Currency> getCurrency(@NotNull String currencyName) {
+        if (!hasCurrency(currencyName)) {
+            return Optional.empty();
+        }
+        UUID currencyId = null;
+        for (String key : BY_NAME.keySet()) {
+            if (key.equalsIgnoreCase(currencyName)) {
+                currencyId = BY_NAME.get(key);
+                break;
+            }
+        }
+        if (currencyId != null) {
+            return getCurrency(currencyId);
+        } else {
+            return Optional.empty();
+        }
+    }
+
+    /**
+     * Returns the {@link Currency} bound to the specified {@link Character} {@code currencyCharacter}, if present.
+     *
+     * @param currencyCharacter the character of the currency needed
+     * @return currency if present, empty optional otherwise
+     * @author MrIvanPlays
+     * @since v1.0.0
+     */
+    public Optional<Currency> getCurrency(@NotNull Character currencyCharacter) {
+        if (!hasCurrency(currencyCharacter)) {
+            return Optional.empty();
+        }
+        return Optional.of(BY_ID.get(BY_CHAR.get(currencyCharacter)));
+    }
+
+    /**
+     * Converts the specified {@code value} from the {@link Currency} {@code from} to the  {@link Currency} {@code to} by
+     * using their {@link Currency#getConversionCoefficient()}.
+     *
+     * @param from  the currency to convert from
+     * @param to    the currency to convert to
+     * @param value the value to convert
+     * @return converted value
+     * @throws IllegalArgumentException if one or both of the currencies are not registered
+     * @author MrIvanPlays, MrNemo64
+     * @since v1.0.0
+     */
+    public double convertCurrency(@NotNull Currency from, @NotNull Currency to, double value) {
```

Comment:
Could also have a 
```java
public double convertCurrency(@NotNull UUID fromId, @NotNull UUID toId, double value) {
  Optional<Currency> from = getCurrency(fromId);
  Optional<Currency> to = getCurrency(toId);
  if(from.isEmpty() || to.isEmpty())
    throw new IllegalArgumentException("some mesage");
  return convertCurrency(from.get(), to.get(), value);
}
```

And maybe instead of value it should be called amount? Dosnt really matter

Reply:
> And maybe instead of value it should be called amount? Dosnt really matter

as you said, doesn't really matter.

Reply:
added.